ee/ucp/kubernetes/layer-7-routing.md

```diff
@@ -19,87 +21,180 @@ Use an ingress controller when you want to:
 
-    The finished YAML should look like this.
+### Step 1: Labeling infrastructure nodes
 
-    ```yaml
-    apiVersion: v1
-    kind: Namespace
-    metadata:
-      name: ingress-nginx
-    ```
-3.  Click **Create**.
-4.  In the **ingress-nginx** namespace, click the **More options** icon,
-    and in the context menu, select **Set Context**.
+To deploy the NGINX controller on the infra nodes, the nodes must be labeled. Identify 2 or 3 nodes as the infra nodes to host the ingress controller. The following example uses three nodes: `dockeree-worker-linux-1`, `dockeree-worker-linux-2`, and `dockeree-worker-linux-3`:
 
-    ![](../images/deploy-ingress-controller-1.png){: .with-border}
+```
+ 🐳  → kubectl get nodes
+NAME                           STATUS    ROLES     AGE       VERSION
+dockeree-worker-linux-1    Ready     <none>    5d        v1.8.11-docker-8d637ae
+dockeree-worker-linux-2    Ready     <none>    5d        v1.8.11-docker-8d637ae
+dockeree-worker-linux-3    Ready     <none>    5d        v1.8.11-docker-8d637ae
 
-## Create a grant
+```
 
-The default service account that's associated with the `ingress-nginx`
-namespace needs access to Kubernetes resources, so create a grant with
-`Restricted Control` permissions.
+The label `infra.role=ingress` is used to label the nodes:
 
-1.  From UCP, navigate to the **Grants** page, and click **Create Grant**.
-2.  Within the **Subject** pane, select **Service Account**. For the
-    **Namespace** select **ingress-nginx**, and select **default** for
-    the **Service Account**. Click **Next**.
-3.  Within the **Role** pane, select **Restricted Control**, and then click
-    **Next**.
-4.  Within the **Resource Set** pane, select the **Type** **Namespace**, and
-    select the **Apply grant to all existing and new namespaces** toggle.
-5.  Click **Create**.
+```
+🐳  → kubectl label node dockeree-worker-linux-1 infra.role=ingress
+node "dockeree-worker-linux-1" labeled
+🐳  → kubectl label node dockeree-worker-linux-2 infra.role=ingress
+node "dockeree-worker-linux-2" labeled
+🐳  → kubectl label node dockeree-worker-linux-3 infra.role=ingress
+node "dockeree-worker-linux-3" labeled
 
-## Deploy NGINX ingress controller
+```
 
-The cluster is ready for the ingress controller deployment, which has three
-main components:
+### Step 2: Create a Dedicated Namespace
 
-- a simple HTTP server, named `default-http-backend`,
-- an ingress controller, named `nginx-ingress-controller`, and
-- a service that exposes the app, named `ingress-nginx`.
+A dedicated namespace, for example, `nginx-controller`, is needed for all infrastructure deployment activities. You also need a service account to enable the ingress controller to work with the Kubernetes API. After creating a namespace and a service account, you must create an RBAC policy to only allow infrastructure deployments on the dedicated nodes that were labelled in the previous example.
```

Comment:
in the configs below, we used `infra` as the namespace. again this is arbitrary. Here you mention nginx-controller. I think you should use `infra` since that's whats in the configs.

Reply:
Thank you - I should have caught that.